src/scripts/push_coral_to_hub.py

```diff
@@ -141,22 +206,101 @@ def main(
+        how="left",
     )
 
+    # Pick a validation set from the training set, by selecting a single recording
+    # from each speaker. We do this by selecting the last recording from each speaker
+    # in the training set, as the first recordings are conversations.
+    validation_recordings = []
+    for speaker_id in train_recordings_df["speaker_id_1"].unique():
+        validation_recordings.append(
+            train_recordings_df[train_recordings_df["speaker_id_1"] == speaker_id].iloc[
+                -1
+            ]
+        )
+    validation_recordings_df = pd.DataFrame(validation_recordings).astype(str)
+
+    # Remove the validation recordings from the training set
+    train_recordings_df = train_recordings_df[
+        ~train_recordings_df["filename"].isin(validation_recordings_df["filename"])
+    ].astype(str)
+
     # Create the dataset
     testset = Dataset.from_pandas(test_recordings_df).cast_column("filename", Audio())
     trainset = Dataset.from_dict(train_recordings_df).cast_column("filename", Audio())
-    dataset_dict = DatasetDict({"train": trainset, "test": testset})
+    validationset = Dataset.from_dict(validation_recordings_df).cast_column(
+        "filename",
+        Audio(),
+    )
+    dataset_dict = DatasetDict(
+        {"train": trainset, "test": testset, "validation": validationset}
+    )
 
     # Create hub-id
     hub_id = f"{hub_id}_v{major_version}.{minor_version}"
 
+    # Check if the dataset already exists on the hub, and if so, prompt the user to
+    # update the version number
+    api = HfApi()
+    try:
+        dataset_info = api.dataset_info(repo_id=hub_id)
+        if dataset_info.id == hub_id:
+            logger.error(
+                (
+                    f"The dataset {hub_id} already exists on the hub!",
+                    "Please update the version number.",
+                )
+            )
+            return
+
+        found_major, found_minor = dataset_info.id.split("_v")[1].split(".")
+        if major_version < int(found_major) or (
+            major_version == int(found_major) and minor_version < int(found_minor)
+        ):
+            logger.error(
+                (
+                    f"The dataset {hub_id} is of a lower version than the one on the",
+                    " hub! Please update the version number.",
+                )
+            )
+            return
+    except RepositoryNotFoundError:
+        if major_version == 1 and minor_version == 0:
+            logger.info(
+                (
+                    f"The dataset {hub_id} doesn't exist on the hub.",
+                    "Creating a new dataset...",
+                )
+            )
+        else:
+            logger.error(
+                (
+                    f"The dataset {hub_id} doesn't exist on the hub!",
+                    "Please create a new dataset with version 1.0.",
+                )
+            )
+            return
```

Comment:
Why is it a requirement to have version 1.0? So it's never allowed to release v1.1, or am I misunderstanding something here?